contracts/contracts/subnet/SubnetActorCheckpointingFacet.sol

```diff
@@ -16,108 +18,93 @@ import {LibGateway} from "../lib/LibGateway.sol";
+        uint64 height = uint64(header.commit.height);
+        // Enforcing a sequential submission
+        ensureValidHeight(height, checkpointStorage.commitmentHeights.signedHeader);
 
-        if (checkpoint.msgs.totalNumMsgs > 0) {
-            LibBottomUpBatch.recordBottomUpBatchCommitment(uint64(checkpoint.blockHeight), checkpoint.msgs);
-        }
-        LibActivity.recordActivityRollup(checkpoint.subnetID, uint64(checkpoint.blockHeight), checkpoint.activity);
+        CometbftLightClient.verifyValidatorsQuorum(header);
 
-        // confirming the changes in membership in the child
-        LibPower.confirmChange(checkpoint.nextConfigurationNumber);
+        checkpointStorage.stateCommitments[height] = StateCommitment({
+            blockHeight: height,
+            commitment: header.header.app_hash
+        });
+        checkpointStorage.commitmentHeights.signedHeader = height;
     }
 
-    /// @notice Checks whether the signatures are valid for the provided signatories and hash within the current validator set.
-    ///         Reverts otherwise.
-    /// @dev Signatories in `signatories` and their signatures in `signatures` must be provided in the same order.
-    ///       Having it public allows external users to perform sanity-check verification if needed.
-    /// @param signatories The addresses of the signatories.
-    /// @param hash The hash of the checkpoint.
-    /// @param signatures The packed signatures of the checkpoint.
-    function validateActiveQuorumSignatures(
-        address[] memory signatories,
-        bytes32 hash,
-        bytes[] memory signatures
-    ) public view {
-        for (uint256 i = 1; i < signatories.length; ) {
-            if (signatories[i] < signatories[i - 1]) {
-                revert SignatureAddressesNotSorted();
-            }
-            if (signatories[i] == signatories[i - 1]) {
-                revert DuplicateValidatorSignaturesFound();
-            }
-
-            unchecked {
-                i++;
-            }
-        }
+    function getLastCommitmentHeights() external view returns (LastCommitmentHeights memory) {
+        return LibCheckpointingStorage.getStorage().commitmentHeights;
+    }
 
-        // This call reverts if at least one of the signatories (validator) is not in the active validator set.
-        uint256[] memory collaterals = s.validatorSet.getTotalPowerOfValidators(signatories);
-        uint256 activeCollateral = s.validatorSet.getTotalActivePower();
+    function recordActivityRollup(
+        uint64 checkpointHeight,
+        SubnetID calldata subnet,
+        CompressedActivityRollup calldata activity,
+        StateCommitmentBreakDown calldata breakdown
+    ) external whenNotPaused {
+        SubnetActorCheckpointingStorage storage checkpointStorage = LibCheckpointingStorage.getStorage();
 
-        uint256 threshold = (activeCollateral * s.majorityPercentage) / 100;
+        validateAppHash(checkpointHeight, breakdown);
```

Comment:
What prevents an arbitrary activity data, unrelated to the commitment breakdown?